plugins/inputs/mqtt_consumer/mqtt_consumer.go

```diff
@@ -115,23 +118,41 @@ func (m *MQTTConsumer) Start(acc telegraf.Accumulator) error {
 		return fmt.Errorf("MQTT Consumer, invalid QoS value: %d", m.QoS)
 	}
 
+	if m.ConnectionTimeout <= 0 {
+		return fmt.Errorf("MQTT Consumer, invalid connection_timeout value: %d", m.ConnectionTimeout)
+	}
+
 	opts, err := m.createOpts()
 	if err != nil {
 		return err
 	}
 
 	m.client = mqtt.NewClient(opts)
+
+	log.Printf("%v", m.connect())
+
+	return nil
+}
+
+func (m *MQTTConsumer) connect() error {
+
```

Comment:
Nitpick, don't put an empty line at the start of the functions.

Reply:
Will do, Thanks!